salt/beacons/__init__.py

```diff
@@ -257,16 +288,23 @@ def add_beacon(self, name, beacon_data):
         data = {}
         data[name] = beacon_data
 
-        if name in self.opts['beacons']:
-            log.info('Updating settings for beacon '
-                     'item: %s', name)
+        if name in self._get_beacons(include_opts=False):
+            comment = 'Cannot update beacon item {0}, ' \
+                      'it is in pillar.'.format(name)
```

Comment:
Perhaps also say that it must be modified directly in the pillar instead of here? This error message is just a little confusing if you haven't read the code itself.